pkg/application/controller/app/action/install.go

```diff
@@ -103,23 +104,29 @@ func Install(ctx context.Context,
 			return nil, err
 		}
 		chartPathBasicOptions.ExistedFile = destfile
-		wait := true
-		if app.Annotations != nil && app.Annotations["ignore-install-wait"] == "true" {
-			wait = false
-		}
-		if app.Labels != nil && app.Labels["application.tkestack.io/type"] == "internal-addon" && time.Now().After(app.CreationTimestamp.Add(5*time.Minute)) {
-			wait = false
+
+		/* provide compatibility with online tke addon apps */
+		if app.Annotations != nil && app.Annotations[applicationprovider.AnnotationProviderNameKey] == "managecontrolplane" {
+			newApp.Spec.Chart.Atomic = true
+			if app.Annotations["ignore-install-wait"] == "true" {
+				newApp.Spec.Chart.Atomic = false
+			}
+			if app.Labels != nil && app.Labels["application.tkestack.io/type"] == "internal-addon" && time.Now().After(app.CreationTimestamp.Add(5*time.Minute)) {
+				newApp.Spec.Chart.Atomic = false
+			}
 		}
+		/* compatibility over, above code need to be deleted atfer the online addon apps are migrated */
+
 		_, err = client.Install(ctx, &helmaction.InstallOptions{
 			Namespace:        newApp.Spec.TargetNamespace,
 			ReleaseName:      newApp.Spec.Name,
 			DependencyUpdate: true,
 			Values:           values,
 			Timeout:          clientTimeOut,
 			ChartPathOptions: chartPathBasicOptions,
-			Wait:             wait,
-			WaitForJobs:      wait,
-			Atomic:           wait,
+			Atomic:           newApp.Spec.Chart.Atomic,
+			Wait:             newApp.Spec.Chart.Atomic,
```

Comment:
wait、waitforjob need exposed for upgrade.go
wait、waitforjob、atomic should be independent